lib/Threading/Pthreads.cpp

```diff
@@ -101,18 +101,26 @@ swift::threading_impl::thread_get_current_stack_bounds() {
 #if defined(__FreeBSD__)
   if (pthread_attr_init(&attr))
     return {};
-#endif
 
-  if (!pthread_getattr_np(pthread_self(), &attr)) {
-    if (!pthread_attr_getstack(&attr, &begin, &size)) {
-      stack_bounds result = { begin, (char *)begin + size };
-      pthread_attr_destroy(&attr);
-      return result;
-    }
+  if (pthread_attr_get_np(pthread_self(), &attr)) {
+    pthread_attr_destroy(&attr);
+    return {};
+  }
+#elif defined(__linux__)
+  if (pthread_getattr_np(pthread_self(), &attr))
+    return {};
+#else
+  // We don't know how to get the thread attr for this platform.
+  return {};
```

Comment:
Missing a `pthread_attr_destroy(&attr);`

Reply:
We haven't called `pthread_attr_init()` on this path, so there's nothing to destroy.

I agree it's a little confusing.  The confusion stems from the fact that FreeBSD's `pthread_attr_get_np()` expects to be passed an allocated `attr`, whereas glibc's `pthread_getattr_np()` wants to allocate one for you.

Reply:
In that case, we're good. I'll approve it and kick off the CI.

Reply:
Thank you!